src/store/index.js

```diff
@@ -36,15 +35,15 @@ export default new Vuex.Store({
   },
   mutations,
   actions: {
-    async updateUserProfile({ commit, state: { address } }) {
-      commit('setUserProfile', await Backend.getProfile(address));
+    async updateUserProfile({ commit, dispatch, state: { address } }) {
+      commit('setUserProfile', await dispatch('backend/getProfile', address));
     },
-    async updatePinnedItems({ commit, state: { address } }) {
-      commit('setPinnedItems', await Backend.getPinnedItems(address));
+    async updatePinnedItems({ commit, dispatch, state: { address } }) {
+      commit('setPinnedItems', await dispatch('backend/getPinnedItems', address));
     },
     async updateCookiesConsent({ commit, dispatch }) {
-      dispatch('backend/callWithAuth', { method: 'getCookiesConsent' })
-        .then((list) => list.forEach(({ scope, status }) => commit('setCookiesConsent', { scope, status: status === 'ALLOWED' })));
+      const payload = await dispatch('backend/callWithAuth', { method: 'getCookiesConsent' });
+      commit('setCookiesConsent', { payload, status: true });
```

Comment:
I agree that you can set all of them at once, but I don't get why `status` need then

Reply:
me too, let's have a call.